packages/integrations/netlify/src/shared.ts

```diff
@@ -10,37 +19,120 @@ export async function createRedirects(
 		if (route.pathname) {
 			if (route.distURL) {
-				_redirects += `
-  ${route.pathname}   /${route.distURL.toString().replace(dir.toString(), '')}    200`;
+				definitions.push({
+					dynamic: false,
+					input: route.pathname,
+					target: prependForwardSlash(route.distURL.toString().replace(dir.toString(), '')),
+					status: 200,
+					weight: 1
+				});
 			} else {
-				_redirects += `
-  ${route.pathname}    /.netlify/${kind}/${entryFile}    200`;
+				definitions.push({
+					dynamic: false,
+					input: route.pathname,
+					target: `/.netlify/${kind}/${entryFile}`,
+					status: 200,
+					weight: 1,
+				});
 
 				if (route.route === '/404') {
-					_redirects += `
-  /*    /.netlify/${kind}/${entryFile}    404`;
+					definitions.push({
+						dynamic: true,
+						input: '/*',
+						target: `/.netlify/${kind}/${entryFile}`,
+						status: 404,
+						weight: 0
+					});
 				}
 			}
 		} else {
 			const pattern =
-				'/' + route.segments.map(([part]) => (part.dynamic ? '*' : part.content)).join('/');
+				'/' + route.segments.map(([part]) => {
+					//(part.dynamic ? '*' : part.content)
+					if(part.dynamic) {
+						if(part.spread) {
+							return '*';
+						} else {
+							return ':' + part.content;
+						}
+					} else {
+						return part.content;
+					}
+				}).join('/');
+
 			if (route.distURL) {
-				_redirects += `
-  ${pattern}   /${route.distURL.toString().replace(dir.toString(), '')}    200`;
+				const target = `${pattern}` + (config.build.format === 'directory' ? '/index.html' : '.html');
+				definitions.push({
+					dynamic: true,
+					input: pattern,
+					target,
+					status: 200,
+					weight: 1
+				});
 			} else {
-				_redirects += `
-  ${pattern}    /.netlify/${kind}/${entryFile}    200`;
+				definitions.push({
+					dynamic: true,
+					input: pattern,
+					target: `/.netlify/${kind}/${entryFile}`,
+					status: 200,
+					weight: 1
+				});
 			}
 		}
 	}
 
+	let _redirects = prettify(definitions);
+
 	// Always use appendFile() because the redirects file could already exist,
 	// e.g. due to a `/public/_redirects` file that got copied to the output dir.
 	// If the file does not exist yet, appendFile() automatically creates it.
 	await fs.promises.appendFile(_redirectsURL, _redirects, 'utf-8');
 }
+
+function prettify(definitions: RedirectDefinition[]) {
+	let minInputLength = 0, minTargetLength = 0;
+	definitions.sort((a, b) => {
+		// Find the longest input, so we can format things nicely
+		if(a.input.length > minInputLength) {
+			minInputLength = a.input.length;
+		}
+		if(b.input.length > minInputLength) {
+			minInputLength = b.input.length;
+		}
+
+		// Same for the target
+		if(a.target.length > minTargetLength) {
+			minTargetLength = a.target.length;
+		}
+		if(b.target.length > minTargetLength) {
+			minTargetLength = b.target.length;
+		}
+
+		// Sort dynamic routes on top
+		if(a.weight > b.weight) {
+			return -1;
+		} else {
+			return 1;
+		}
```

Comment:
I don't think this matters from some quick tests, but maybe we can make this
```suggestion
		return b.weight - a.weight;
```

so that 2 equal weights return 0 so it retains the order.